website/docs/r/vapp_access_control.html.markdown

```diff
@@ -0,0 +1,135 @@
+---
+layout: "vcd"
+page_title: "vCloudDirector: vcd_vapp_access_control"
+sidebar_current: "docs-vcd-resource-vapp-access-control"
+description: |-
+  Provides a vCloud Director Access Control structure for a vApp.
+---
+
+# vcd\_vapp\_access\_control
+
+Provides a vCloud Director Access Control structure for a vApp. This can be used to create, update, and delete access control structures for a vApp.
+
+!> **Warning:** The access control info is tied to a vApp. Thus, there could be only one instance per vApp. Using a different
+definition for the same vApp ID will result in a previous instance to be overwritten.
+
+Supported in provider *v2.10+*
+
+## Example Usage
+
+```hcl
+
+data "vcd_org_user" "ac-admin1" {
+  name = "ac-admin1"
+}
+
+data "vcd_org_user" "ac-vapp-creator2" {
+  name = "ac-vapp-creator2"
+}
+
+data "vcd_vapp" "Vapp-AC-0" {
+  name = "Vapp-AC-0"
+}
+
+data "vcd_vapp" "Vapp-AC-1" {
+  name = "Vapp-AC-1"
+}
+
+data "vcd_vapp" "Vapp-AC-2" {
+  name = "Vapp-AC-2"
+}
+
+resource "vcd_vapp_access_control" "AC-not-shared" {
+
+  vapp_id  = data.vcd_vapp.Vapp-AC-0.id
+
+  shared_to_everyone    = false
+}
+
+
+resource "vcd_vapp_access_control" "AC-global" {
+
+  vapp_id  = data.vcd_vapp.Vapp-AC-1.id
+
+  shared_to_everyone    = true
+  everyone_access_level = "Change"
+}
+
+resource "vcd_vapp_access_control" "AC-users" {
+  vapp_id  = data.vcd_vapp.Vapp-AC-2.id
+
+  shared_to_everyone    = false
+
+  shared {
+    user_id      = data.vcd_org_user.ac-admin1.id
+    access_level = "FullControl"
+  }
+  shared {
+    user_id      = data.vcd_org_user.ac-vapp-creator2.id
+    access_level = "Change"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to which the vApp belongs. Optional if defined at provider level.
+* `vdc` - (Optional) The name of organization to which the vApp belongs. Optional if defined at provider level.
+* `vapp_id` - (Required) A unique identifier for the vApp.
+* `shared_to_everyone` - (Required) Whether the vApp is shared to everyone
```

Comment:
We need to write up how does this field conflict/behave if both it and `shared` below are set.

Reply:
done